src/core/parser.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <initializer_list>
+#include <nlohmann/json.hpp>
+#include "parse_feeder.hpp"
+#include "parse_worker.hpp"
+#include "comments_parser.hpp"
+#include "ignore_parser.hpp"
+
+namespace docgen {
+namespace core {
+
+class Parser : private ParseWorker
+{
+	public:
+		Parser()
+			: ParseWorker {
+				CommentParser(Symbol::BEGIN_LINE_COMMENT, Symbol::NEWLINE),
+				CommentParser(Symbol::BEGIN_BLOCK_COMMENT, Symbol::END_BLOCK_COMMENT, {
+					IgnoreParser(Symbol::NEWLINE, Symbol::STAR)
+				})
+			}
+		{}
+
+		void process(const std::vector<token_t>& tokens);
+
+		const nlohmann::json& parsed() const { return parsed_; }
+
+	private:
+		nlohmann::json parsed_;
```

Comment:
Instead of owning a `parsed_` object, how about owning a `ParseFeeder` object. The `parsed` function returns `parsed_` inside `ParseFeeder` object. See my comments about `ParseFeeder` - I want to make it a non-static class.